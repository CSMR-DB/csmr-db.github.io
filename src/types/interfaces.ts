import { ColorProperty } from 'csstype'
import { FluidObject } from 'gatsby-image'

/**
 * SiteMetadata GraphQL data
 */
export interface ISiteMetadata {
  site: {
    siteMetadata: {
      title: string
      description: string
      author: string
      image: string
      url: string
    }
  }
}

/**
 * Generic GraphQL Data
 */
export interface IMarkdownRemark<T> {
  frontmatter: T
  html: string
  timeToRead: number
}

export interface ISingleItemData<T> {
  data: {
    markdownRemark: IMarkdownRemark<T>
  }
}

export interface IAllMarkdownRemark<T> {
  edges: IEdge<T>[]
}

export interface IListItemData<T> {
  data: {
    allMarkdownRemark: IAllMarkdownRemark<T>
  }
}

export interface IEdge<T> {
  node: INode<T>
}

export interface INode<T> {
  frontmatter: T
  wordCount: {
    words: number
    sentences: number
    paragraphs: number
  }
  timeToRead: number
  html: string
}

/**
 * Experience GraphQL Data
 */
export interface IExperienceFrontmatter {
  title: string
  dateStart: string
  dateEnd: string
  description: string
  backgroundColor: string
  type: string
  excerpt: string
}

export interface IExperienceEdge {
  node: {
    frontmatter: IExperienceFrontmatter
    wordCount: {
      words: number
      sentences: number
      paragraphs: number
    }
    html: string
  }
}

export interface IExperienceAllMarkdownRemark {
  edges: IExperienceEdge[]
}

/**
 * Skillset GraphQL data
 */
export interface ISkillsetFrontmatter {
  favorite: boolean | null
  title: string
  date: string
  path: string
  excerpt: string
  tags?: string[]
  time?: number
  level?: number
  skillColor?: ColorProperty
}

export interface ISkillsetEdge {
  node: {
    frontmatter: ISkillsetFrontmatter
    wordCount: {
      words: number
      sentences: number
      paragraphs: number
    }
    html: string
  }
}

export interface ISkillsetAllMarkdownRemark {
  edges: ISkillsetEdge[]
}

/**
 * Projects GraphQL data
 */
export interface IProjectFrontmatter {
  favorite: boolean | null
  title: string
  category: 'Programming' | 'Graphic Design'
  date: string
  path: string
  excerpt: string
  featuredImage?: IImageSharp
  featuredVideo?: string
  tags?: string[]
}

export interface IProjectsEdge {
  node: {
    frontmatter: IProjectFrontmatter
    wordCount: {
      words: number
      sentences: number
      paragraphs: number
    }
    html: string
  }
}

export interface IProjectsAllMarkdownRemark {
  edges: IProjectsEdge[]
}

/**
 * Image GraphQL data
 */
export interface IImageSharp {
  name: string
  relativePath: string
  childImageSharp: {
    fluid: FluidObject
  }
}

export interface IImageSharpEdge {
  node: IImageSharp
}

export interface IImageSharpAllFiles {
  edges: IImageSharpEdge[]
}
