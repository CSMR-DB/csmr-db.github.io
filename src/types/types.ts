import React from 'react'

export type AnyExoticRefComponent<T> = React.ForwardRefExoticComponent<
  T & React.RefAttributes<any>
>

export type AnyExoticRefTargets =
  | ((instance: any) => void)
  | React.MutableRefObject<any>
  | null
