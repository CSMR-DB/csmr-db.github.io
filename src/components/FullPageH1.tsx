import styled, { StyledComponent } from 'styled-components'

export const FullPageH1: StyledComponent<'h1', any, {}, never> = styled.h1`
  margin-top: 4rem;
`
