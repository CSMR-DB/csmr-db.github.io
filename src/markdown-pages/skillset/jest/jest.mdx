---
category: 'Programming'
level: 2
skillColor: '#C63D14'
time: 400
title: 'Jest'
excerpt: "I mean, it's one of many testing frameworks. I mainly just like the logo."
icon: <JestSVG/>
---
