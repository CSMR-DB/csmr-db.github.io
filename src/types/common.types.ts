export type StringOrUrlObject = string | { url: string; text: string }

export type StringOrUrlArray = string | StringOrUrlObject[]
