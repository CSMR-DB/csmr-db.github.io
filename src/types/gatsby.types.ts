import { Reporter } from 'gatsby'

export type AllMdxEdge = {
  node: {
    frontmatter: {
      title: string
      tags?: string[]
    }
    generatedRoute: string
  }
}

export type AllMdxData = {
  allMdx: {
    edges: AllMdxEdge[]
  }
}

export type CreatePagesArgs = {
  actions: {
    createPage: (args: { path: string; component: string; context: {} }) => void
  }
  graphql: (
    args: string
  ) => {
    errors?: Error[]
    data: AllMdxData
  }
  reporter: Reporter
}

export type CreateSchemaCustomizationArgs = {
  actions: {
    createTypes: (args: string) => void
  }
  createResolvers: (args: {}) => void
}
