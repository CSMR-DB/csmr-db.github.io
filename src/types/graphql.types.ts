import { ColorProperty } from 'csstype'
import { FluidObject } from 'gatsby-image'

/**
 * SiteMetadata GraphQL data
 */
export type SiteMetadata = {
  title: string
  description: string
  author: string
  image: string
  url: string
}

type FrontmatterBase = {
  title: string
  excerpt: string
}

export type MarkdownRemark<T extends FrontmatterBase> = {
  generatedRoute: string
  frontmatter: T
  body?: string
  wordCount?: {
    words?: number
    sentences?: number
    paragraphs?: number
  }
  timeToRead?: number
}

export type MarkdownEdge<T extends FrontmatterBase> = {
  node: MarkdownRemark<T>
}

export type AllMarkdownRemark<T extends FrontmatterBase> = {
  edges: MarkdownEdge<T>[]
}

export type FrontmatterCategories = 'Programming' | 'Graphic Design'

/**
 * Experience GraphQL Frontmatter
 */
export type ExperienceFrontmatter = {
  title: string
  dateStart: string
  dateEnd: string
  description: string
  backgroundColor: string
  type: string
  excerpt: string
}

export type ExperienceAllMarkdownRemark = AllMarkdownRemark<
  ExperienceFrontmatter
>
export type ExperienceMarkdownEdge = MarkdownEdge<ExperienceFrontmatter>
export type ExperienceMarkdownRemark = MarkdownRemark<ExperienceFrontmatter>

/**
 * Skillset GraphQL Frontmatter
 */
export type SkillsetFrontmatter = {
  favorite: boolean | null
  title: string
  category: FrontmatterCategories
  date: string
  excerpt: string
  tags?: string[]
  time?: number
  level?: number
  skillColor?: ColorProperty
  icon?: string
}

export type SkillsetAllMarkdownRemark = AllMarkdownRemark<SkillsetFrontmatter>
export type SkillsetMarkdownEdge = MarkdownEdge<SkillsetFrontmatter>
export type SkillsetMarkdownRemark = MarkdownRemark<SkillsetFrontmatter>

/**
 * Projects GraphQL Frontmatter
 */
export type ProjectFrontmatter = {
  favorite: boolean | null
  title: string
  category: FrontmatterCategories
  date: string
  excerpt: string
  featuredImage?: FileChildImageSharp
  featuredVideo?: string
  tags?: string[]
}

export type ProjectsAllMarkdownRemark = AllMarkdownRemark<ProjectFrontmatter>
export type ProjectMarkdownEdge = MarkdownEdge<ProjectFrontmatter>
export type ProjectMarkdownRemark = MarkdownRemark<ProjectFrontmatter>

/**
 * File GraphQL data
 */
export type FileNode = {
  name: string
  relativePath: string
}

export type FileEdge<T extends FileNode> = {
  node: T
}

export type AllFile<T extends FileNode> = {
  edges: FileEdge<T>[]
}

export type FileChildImageSharp = {
  name: string
  relativePath: string
  childImageSharp: {
    fluid: FluidObject
  }
}

export type ImageSharpAllFiles = AllFile<FileChildImageSharp>
export type ImageSharpEdge = FileEdge<FileChildImageSharp>
