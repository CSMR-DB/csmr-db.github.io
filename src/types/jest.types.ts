import { RenderResult } from '@testing-library/react'
import { ReactTestRendererJSON } from 'react-test-renderer'

export type RTRTree = ReactTestRendererJSON | ReactTestRendererJSON[] | null

export type RenderResultImport = RenderResult<
  typeof import('c:/Users/casim/Documents/Code/__nodejs/csmr-db/node_modules/@testing-library/dom/types/queries')
>

export type TestAttrs = {
  'data-testid'?: string
}
