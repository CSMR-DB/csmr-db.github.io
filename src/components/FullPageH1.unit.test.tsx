import React from 'react'
import { create, ReactTestRenderer } from 'react-test-renderer'

import { RTRTree } from '../types/jest.types'

import { FullPageH1 } from './FullPageH1'

describe('Layout', (): void => {
  it('should have the following style rules', (): void => {
    const comp: ReactTestRenderer = create(
      <FullPageH1>I am a big H1</FullPageH1>
    )
    const json: RTRTree = comp.toJSON()

    expect(comp).toMatchSnapshot()

    expect(json).toHaveStyleRule('margin-top', '4rem')
  })
})
