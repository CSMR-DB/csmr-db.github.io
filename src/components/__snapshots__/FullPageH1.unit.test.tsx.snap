// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should have the following style rules 1`] = `
.c0 {
  margin-top: 4rem;
}

<h1
  className="c0"
>
  I am a big H1
</h1>
`;
